### ISSUE TITLE
[JENKINS-19828] Kill or rework auto refresh

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-15#3641]
<p>The <tt>auto_refresh</tt> query parameter (and associated <tt>hudson_auto_refresh</tt> cookie) not only wreaks havoc on forms or any other pages that forget to add <tt>norefresh=true</tt>, it can cause serious performance issues in large Jenkins installations if some team members naïvely turn on auto refresh on the dashboard or some other expensive page and then go home for the night. Options:</p>

<p>1. Delete this (mis)feature, adding back in more AJAXy behaviors to various pages upon request.<br/>
2. Allow a Jenkins administrator to disable it for a given installation.<br/>
3. Keep it, but make the <tt>Refresh</tt> time (<tt>Functions.configureAutoRefresh</tt>) increase exponentially after each automatic refresh, resetting to 10s only after an explicit page load (if these can be somehow distinguished).</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick">jglick</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-19828" target="_blank">Kill or rework auto refresh</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: core
<li><b>label(s)</b>: 2.0-rejected, auto-refresh, performance
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2020-08-31T16:35:01+00:00
<li><b>votes</b>: 15
<li><b>watchers</b>: 13
<li><b>imported</b>: 2025-11-16
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>The <tt>auto_refresh</tt> query parameter (and associated <tt>hudson_auto_refresh</tt> cookie) not only wreaks havoc on forms or any other pages that forget to add <tt>norefresh=true</tt>, it can cause serious performance issues in large Jenkins installations if some team members naïvely turn on auto refresh on the dashboard or some other expensive page and then go home for the night. Options:</p>

<p>1. Delete this (mis)feature, adding back in more AJAXy behaviors to various pages upon request.
2. Allow a Jenkins administrator to disable it for a given installation.
3. Keep it, but make the <tt>Refresh</tt> time (<tt>Functions.configureAutoRefresh</tt>) increase exponentially after each automatic refresh, resetting to 10s only after an explicit page load (if these can be somehow distinguished).</p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> oleg_nenashev's <a href="https://issues.jenkins.io/browse/JENKINS-19828?focusedId=205271&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205271">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>&gt; Isn't "1" why we're discussing Servlet 3 to get SSE?</p>

<p>I suppose this is a question to Jesse. Could you give a link to this discussion?</p>

<p>Additional approach:
4. Alter the default behavior in layout.jelly (disable autorefresh if norefresh is unspecified)</p>
</pre>
</details>

</details>
<p>> Isn't "1" why we're discussing Servlet 3 to get SSE?</p>

<p>I suppose this is a question to Jesse. Could you give a link to this discussion?</p>

<p>Additional approach:<br/>
4. Alter the default behavior in layout.jelly (disable autorefresh if norefresh is unspecified)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-19828?focusedId=210491&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-210491">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Not sure which to resolve as Duplicate here. While the other is older, this one has more (bad) options.</p>
</pre>
</details>

</details>
<p>Not sure which to resolve as Duplicate here. While the other is older, this one has more (bad) options.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/dam321.png" title="dam321's avatar" /> dam321's <a href="https://issues.jenkins.io/browse/JENKINS-19828?focusedId=238796&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-238796">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>how can we turn off auto refresh ...as we do not have aut refresh enabled but our jenkins instace keeps refreshing every 20 seconds all pages in jenkins does the same.
version 1.624</p>
</pre>
</details>

</details>
<p>how can we turn off auto refresh ...as we do not have aut refresh enabled but our jenkins instace keeps refreshing every 20 seconds all pages in jenkins does the same.<br/>
version 1.624</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick's <a href="https://issues.jenkins.io/browse/JENKINS-19828?focusedId=238802&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-238802">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dam321" class="user-hover" rel="dam321">dam321</a> please take it to the Jenkins users’ list rather than adding noise to the JIRA issue.</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dam321" class="user-hover" rel="dam321">dam321</a> please take it to the Jenkins users’ list rather than adding noise to the JIRA issue.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/batmat.png" title="batmat's avatar" /> batmat's <a href="https://issues.jenkins.io/browse/JENKINS-19828?focusedId=270504&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-270504">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>As a user, it's been years I always disable this. The few times I've used it, it's screwed a form I was filling in and so on. 
So we should indeed either:</p>

<ul>
	<li>simply kill it in the main code,</li>
	<li>or offer a simple way for people to simply disable it totally. Like Oleg says: if the UI has a value of 0 e.g., just do not display the field at all. We could probably even use it as a progressive strategy to kill it: do that, plus define the default to 0 for new installs? And in the meantime, make a poll about that on the users list to take the temperature?</li>
</ul>

</pre>
</details>

</details>
<p>As a user, it's been years I always disable this. The few times I've used it, it's screwed a form I was filling in and so on. <br/>
So we should indeed either:</p>

<ul>
	<li>simply kill it in the main code,</li>
	<li>or offer a simple way for people to simply disable it totally. Like Oleg says: if the UI has a value of 0 e.g., just do not display the field at all. We could probably even use it as a progressive strategy to kill it: do that, plus define the default to 0 for new installs? And in the meantime, make a poll about that on the users list to take the temperature?</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ssbarnea.png" title="ssbarnea's avatar" /> ssbarnea's <a href="https://issues.jenkins.io/browse/JENKINS-19828?focusedId=285295&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-285295">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Why was this rejected from 2.0? This "feature" is only creating bugs and should not exist at all. There are other ways to implement refresh whenever this is needed, a generic page refresh is a disaster for so many config pages. </p>

<p>A web page should never reload itself completely, it does it usually points to bad implementation. I guess the reason why this feature-bug exists is only because it is older than jQuery itself.</p>
</pre>
</details>

</details>
<p>Why was this rejected from 2.0? This "feature" is only creating bugs and should not exist at all. There are other ways to implement refresh whenever this is needed, a generic page refresh is a disaster for so many config pages. </p>

<p>A web page should never reload itself completely, it does it usually points to bad implementation. I guess the reason why this feature-bug exists is only because it is older than jQuery itself.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-19828?focusedId=285313&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-285313">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<blockquote><p>Why was this rejected from 2.0?</p></blockquote>

<p>2.0 was a specific release 9 months ago and already was behind schedule with the changes that did make it in. This was simply out of scope for that. And given the traction Blue Ocean has I doubt we'll ever develop a replacement in core, the best I expect we'll do is kill it off. However…</p>

<blockquote><p>This "feature" is only creating bugs and should not exist at all. There are other ways to implement refresh whenever this is needed, a generic page refresh is a disaster for so many config pages.</p></blockquote>

<p>It's opt in. And apparently, the people who do that don't want to go without it, despite all the brokenness.</p>
</pre>
</details>

</details>
<blockquote><p>Why was this rejected from 2.0?</p></blockquote>

<p>2.0 was a specific release 9 months ago and already was behind schedule with the changes that did make it in. This was simply out of scope for that. And given the traction Blue Ocean has I doubt we'll ever develop a replacement in core, the best I expect we'll do is kill it off. However…</p>

<blockquote><p>This "feature" is only creating bugs and should not exist at all. There are other ways to implement refresh whenever this is needed, a generic page refresh is a disaster for so many config pages.</p></blockquote>

<p>It's opt in. And apparently, the people who do that don't want to go without it, despite all the brokenness.</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Duplicates: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-15/issues?q=in%3Atitle%20JENKINS-484">JENKINS-484</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Originally related to: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-15/issues?q=in%3Atitle%20JENKINS-10583">JENKINS-10583</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Epic: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-15/issues?q=in%3Atitle%20JENKINS-31156%20label%3Aepic">JENKINS-31156</a>]</i>
